ext/openssl/openssl.c

```diff
@@ -1402,7 +1518,8 @@ PHP_FUNCTION(openssl_x509_export_to_file)
 		php_openssl_store_errors();
 		php_error_docref(NULL, E_WARNING, "Error opening file %s", filename);
 	}
-	if (Z_TYPE_P(zcert) != IS_RESOURCE) {
+
+	if (cert_str) {
 		X509_free(cert);
```

Comment:
Yeah it's better handled but not sure how it could help in this case as it would have to be freed anyway, right?

Reply:
A lot of code in openssl is dealing with conditionally freeing certs depending on where they come from. OpenSSL 1.1 would allow us to just use X509_up_ref() if it's an existing cert, and then X509_free() everything unconditionally, removing these "if (cert_str)" and similar checks. But well, we're not there yet :)

Reply:
Yeah it would be definitely useful and I think we should actually start using it for cloning. In 1.0.2 we could just dup it instead.

Just to clarify what I meant by this case is that this is just when the X509 is allocated from string so it would stay the same as it is just one off usage that can't be stored anywhere.